lib/rules/no-constant-condition.js

```diff
@@ -16,6 +16,27 @@ module.exports = function(context) {
     // Helpers
     //--------------------------------------------------------------------------
 
+
+    /**
+     * Checks if a branch node of LogicalExpression short circuits the whole condition
+     * @param {ASTNode} node The branch of main condition which needs to be checked
+     * @param {string} operator The operator of the main LogicalExpression.
+     * @returns {boolean} true when condition short circuits whole condition
+     */
+    function isShortCircuit(node, operator) {
```

Comment:
The term "short circuit" is not appropriate here. Short circuiting means the operator doesn't evaluate the right operand for some values of the left operand. This function is checking for the use of a logical identity.